pyanaconda/modules/network/nm_client.py

```diff
@@ -672,7 +672,7 @@ def bound_hwaddr_of_device(nm_client, device_name, ifname_option_values):
     :return: hwaddress of the device if bound, or None
     :rtype: str or None
     """
-    for ifname_value in ifname_option_values:
+    for ifname_value in ifname_option_values or []:
```

Comment:
I would replace `ifname_option_values` with an empty list in `update_connection_from_ksdata` to make sure it is always a list. The current workaround in `bound_hwaddr_of_device` is a little hard to follow.

Reply:
Done.